File: minimum-difference-pair/MinimumDifferencePair.js
function minimumDifference(array) {
    array.sort((a, b) => a - b);
    let results = [];
    let difference = array[1] - array[0];
    
    for(let i = 2; i < array.length; i++) {
        difference = Math.min(difference, (array[i] - array[i - 1]));
    }
    
    for(let j = 1; j < array.length; j++) {
        if(difference === (array[j] - array[j - 1])) {
            results.push([array[j], array[j - 1]]);
        }
    }
    return results;
}

exports.minimumDifference = minimumDifference;

File: minimum-difference-pair/MinimumDifferencePair.swift
import Foundation

class MinimumDifference {
    public func minimumDifference(array: inout [Int]) -> [[Int]] {
        array.sort()
        var difference = array[1] - array[0]
        var results: [[Int]] = []
        for i in stride(from: 2, to: array.count, by: 1) {
            difference = min(difference, array[i] - array[i - 1])
        }
        for j in stride(from: 1, to: array.count, by: 1) {
            if(difference == array[j] - array[j - 1]) {
                results.append([array[j], array[j - 1]])
            }
        }
        return results;
    }
}